File: serialcomms.cpp
/***

Embed microcontroller serial port listener file
that shall listen to the host PC and print out
strings sent from the host PC.

***/
#include "mbed.h"              
 
Serial pc(USBTX, USBRX);
 
int main() {
    while(1) {
        /* Listen for characters sent
        from host PC and return the chracters
        through the serial port via USB back
        to the host.
        */
        pc.putc(pc.getc());
    }
}

File: serialcomms.py
'''
Communicates with Mbed LPC1768
microcontroller via usb serialport.
Indefinitely waits for user input to
send through serial port and reads from
the port the response from the microcontroller.
'''
import serial
# Establish connection to the Mbed unit.
ser = serial.Serial('/dev/ttyACM0',
        9600, timeout=2)


while True:
    serial_input = raw_input(
            "Enter string to send to microcontroller: ")
    ser.write(serial_input + "\n")
    print "Sent string: %s" % serial_input
    print "Response: %s" % ser.readline()
